### PR TITLE
fix(ios): enable swift development in Titanium

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-28062

### CODE REVIEW

**Comment by vijaysingh-axway** [comment]
_build.js changes are from [here](https://github.com/appcelerator/titanium_mobile/pull/11689/commits/cd6f844936bb8a36e6225905a84fe0eb172b7c84) with minor change.

**Comment by sgtcoolguy** [comment]
@vijaysingh-axway How do we verify this change?

**Comment by vijaysingh-axway** [comment]
@sgtcoolguy  I updated 2 unit tests Ti.Platform.id and Ti.Platform.macaddress, which is dependent on a function `func appIdentifier() -> String?` (extension of TiUtils) which is defined in TiUtils+Addons.swift. If this unit test is successful, which means the swift development is enabled in titanium.

File: iphone/TitaniumKit/TitaniumKit.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1010,7 +1010,7 @@
 					"$(PROJECT_DIR)/TitaniumKit/Libraries/APSAnalytics",
 					"$(PROJECT_DIR)/TitaniumKit/Libraries/APSHTTPClient",
 				);
-				ONLY_ACTIVE_ARCH = NO;
+				ONLY_ACTIVE_ARCH = YES;
```
Comment (sgtcoolguy) [inline iphone/TitaniumKit/TitaniumKit.xcodeproj/project.pbxproj:1013]: Is this change intended?

File: iphone/cli/commands/_build.js [hunk 2]
```diff
@@ -2956,6 +2956,8 @@ iOSBuilder.prototype.createXcodeProject = function createXcodeProject(next) {
 				obj.path = xobjs.PBXFileReference[id + '_comment'] = '"' + appName + '.entitlements"';
 			} else if (obj.path === 'Titanium.app') {
 				obj.path = xobjs.PBXFileReference[id + '_comment'] = '"' + appName + '.app"';
+			}	else if (obj.path === '"Titanium-Bridging-Header.h"') {
+				obj.path = xobjs.PBXFileReference[id + '_comment'] = '"' + scrubbedAppName + '-Bridging-Header.h"';
```
Comment (sgtcoolguy) [inline iphone/cli/commands/_build.js:2960]: I find the ES6+ style string concat a little easier on the eyes:
```suggestion
				obj.path = xobjs.PBXFileReference[id + '_comment'] = `"${scrubbedAppName}-Bridging-Header.h"`;
```

File: iphone/cli/commands/_build.js [hunk 3]
```diff
@@ -4618,6 +4625,11 @@ iOSBuilder.prototype.copyTitaniumiOSFiles = function copyTitaniumiOSFiles() {
 		path.join(this.platformPath, 'iphone', 'Titanium_Prefix.pch'),
 		path.join(this.buildDir, name + '_Prefix.pch')
 	);
+	copyAndReplaceFile.call(
+		this,
+		path.join(this.platformPath, 'iphone', 'Titanium-Bridging-Header.h'),
+		path.join(this.buildDir, name + '-Bridging-Header.h')
```
Comment (sgtcoolguy) [inline iphone/cli/commands/_build.js:4631]:
```suggestion
		path.join(this.buildDir, `${name}-Bridging-Header.h`)
```

File: tests/Resources/ti.platform.test.js [hunk 4]
```diff
@@ -215,7 +215,13 @@ describe('Titanium.Platform', function () {
 
 	it('.id', () => {
 		should(Ti.Platform).have.readOnlyProperty('id').which.is.a.String();
-		// TODO Verify format?!
+		if (OS_IOS) {
+			const platformId = Ti.Platform.id;
+			should(platformId).be.a.String();
+			should(platformId.length).eql(36);
+			// Verify format using regexp!
+			should(platformId.match(/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i)).not.eql(null);
```
Comment (sgtcoolguy) [inline tests/Resources/ti.platform.test.js:223]: should.js already has a method for matching regexes:
```suggestion
			platformId.should.match(/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i);
```

File: tests/Resources/ti.platform.test.js [hunk 5]
```diff
@@ -225,6 +231,13 @@ describe('Titanium.Platform', function () {
 
 	it('.macaddress', () => {
 		should(Ti.Platform).have.readOnlyProperty('macaddress').which.is.a.String();
+		if (OS_IOS) {
+			const platformId = Ti.Platform.id;
+			should(platformId).be.a.String();
+			should(platformId.length).eql(36);
+			// Verify format using regexp!
+			should(platformId.match(/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i)).not.eql(null);
```
Comment (sgtcoolguy) [inline tests/Resources/ti.platform.test.js:239]: The test here is for the `macaddress` property, but you're checking the `Ti.Platform.id` property.

**Review by sgtcoolguy — CHANGES_REQUESTED** [review]

File: iphone/TitaniumKit/TitaniumKit.xcodeproj/project.pbxproj (hunk 1 above)
Comment (vijaysingh-axway) [reply iphone/TitaniumKit/TitaniumKit.xcodeproj/project.pbxproj:1013]: Yes. This is for 'Debug' mode. A minor change also required in scheme of Distribution target. Updated that as well.

File: iphone/cli/commands/_build.js (hunk 2 above)
Comment (vijaysingh-axway) [reply iphone/cli/commands/_build.js]: Updated.

File: tests/Resources/ti.platform.test.js (hunk 5 above)
Comment (vijaysingh-axway) [reply tests/Resources/ti.platform.test.js]: Copy paste issue :). Fixed.

**Review by sgtcoolguy — APPROVED** [review]

**Comment by gvfbossa** [comment]
Hi! How can I download this 9.2.0 build? When I click on it to download, a Jenkins page opens and I get the message:
"gvfbossa is missing the Overall/Read permission"
